frappe/email/doctype/email_queue/test_email_queue.py

```diff
@@ -39,3 +40,38 @@ def test_email_queue_deletion_based_on_modified_date(self):
 
 		self.assertTrue(frappe.db.exists("Email Queue", new_record.name))
 		self.assertTrue(frappe.db.exists("Email Queue Recipient", {"parent": new_record.name}))
+
+	def test_failed_email_notification(self):
+		subject = frappe.generate_hash()
+		email_record = frappe.new_doc("Email Queue")
+		email_record.sender = "Test <test@example.com>"
+		email_record.message = f"""MIME-Version: 1.0
+Message-Id: {frappe.generate_hash()}
+X-Original-From: Test <test@example.com>
+Subject: {subject}
+From: Test <test@example.com>
+To: <!--recipient-->
+Date: {frappe.utils.now_datetime().strftime('%a, %d %b %Y %H:%M:%S %z')}
+Reply-To: test@example.com
+X-Frappe-Site: {frappe.local.site}
+"""
+		email_record.status = "Error"
+		email_record.retry = get_email_retry_limit()
+		email_record.priority = 1
+		email_record.reference_doctype = "User"
+		email_record.reference_name = "Administrator"
+		email_record.insert()
+
+		# Simulate an exception so that we get a notification
+		try:
+			with SendMailContext(queue_doc=email_record):
+				raise Exception("Test Exception")
+		except Exception:
+			pass
+
+		notification_log = frappe.db.get_value(
+			"Notification Log",
+			{"subject": f"Failed to send email with subject: {subject}"},
+			fieldname="*",
+		)
+		assert notification_log is not None
```

Comment:
```suggestion
		self.assertTrue(notification_log)
```

This would work if we were using pytest :grimacing:

Reply:
Ah, judging by method name I had initially assumes it just checks for true explicitly, just saw its not that :P switched